ibis/expr/types/logical.py

```diff
@@ -48,21 +48,184 @@ def ifelse(
         return ops.Where(self, true_expr, false_expr).to_expr()
 
     def __and__(self, other: BooleanValue) -> BooleanValue:
+        """Construct a binary AND conditional expression with `self` and `other`.
+
+
+        Parameters
+        ----------
+        self
+            Left operand
+        other
+            Right operand
+
+        Returns
+        -------
+        BooleanValue
+            A Boolean expression
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [[1], [], [42, 42], None]})
+        >>> t.arr.contains(42).__and__(t.arr.contains(1))
```

Comment:
```suggestion
        >>> t.arr.contains(42) & t.arr.contains(1)
```

It's very uncommon for users to call "dunder" methods directly.